src/sys/socket/mod.rs

```diff
@@ -803,6 +803,38 @@ pub enum ControlMessageOwned {
     #[cfg(any(target_os = "linux"))]
     TlsGetRecordType(TlsGetRecordType),
 
+    #[cfg(any(
+        apple_targets,
+        linux_android,
+        target_os = "freebsd",
+    ))]
+    #[cfg(feature = "net")]
+    IpTos(libc::c_int),
+
+    #[cfg(any(
+        apple_targets,
+        linux_android,
+        target_os = "freebsd",
+    ))]
+    #[cfg(feature = "net")]
+    Ipv6TClass(libc::c_int),
+
+    #[cfg(any(
+        apple_targets,
+        linux_android,
+        target_os = "freebsd",
+    ))]
+    #[cfg(feature = "net")]
+    IpTtl(libc::c_int),
+
+    #[cfg(any(
+        apple_targets,
+        linux_android,
+        target_os = "freebsd",
+    ))]
+    #[cfg(feature = "net")]
+    Ipv6HopLimit(libc::c_int),
```

Comment:
This variant is gated with
```rs
    #[cfg(any(linux_android, freebsdlike, apple_targets, target_os = "haiku"))]
```

 in `ControlMessage`, seems that we should also gate it in this way here?